### DISCUSSION TITLE
With `set_toolchains("@cosmocc")` :  fatal error: string: No such file or directory

### DISCUSSION THREAD

**Discussion by raphael10-collab** [xmake-io/xmake#5262] [Q&A]
With this xmake.lua file : 

    set_languages("c99", "c++20")
    add_rules("mode.debug", "mode.release")
    
    add_requires("cosmocc")
    
    target("isdigits")
        set_kind("binary")
        add_files("/home/raphy/isdigits/src/*.cpp")
        --set_toolchains("@cosmocc")

It works fine : 

    raphy@raohy:~$ xmake f -o /home/raphy/isdigits/builddir
    
    raphy@raohy:~$ xmake
    [ 50%]: cache compiling.release isdigits/src/isdigits.cpp
    [ 75%]: linking.release isdigits
    [100%]: build ok, spent 0,482s
    
    raphy@raohy:~$ ./isdigits/builddir/linux/x86_64/release/isdigits 
    Only digits
    
    Alpha Numerical

If I activate this line :  `set_toolchains("@cosmocc")`  : 
    
    set_languages("c99", "c++20")
    add_rules("mode.debug", "mode.release")
    
    add_requires("cosmocc")
    
    target("isdigits")
        set_kind("binary")
        add_files("/home/raphy/isdigits/src/*.cpp")
        set_toolchains("@cosmocc")

I get that error : 

    raphy@raohy:~$ xmake
    [ 50%]: cache compiling.release isdigits/src/isdigits.cpp
    error: isdigits/src/isdigits.cpp:3:10: fatal error: string: No such file or directory
        3 | #include <string>
          |          ^~~~~~~~
    compilation terminated.
      > in isdigits/src/isdigits.cpp

Output of `xmake -v` : 

    raphy@raohy:~$ xmake -v
    [ 50%]: cache compiling.release isdigits/src/isdigits.cpp
    /home/raphy/.xmake/packages/c/cosmocc/3.4.0/7e5513266cda47b5b64419a0b92403ee/bin/cosmocc -c -fvisibility=hidden -DNDEBUG -o isdigits/builddir/.objs/isdigits/linux/x86_64/release/isdigits/src/isdigits.cpp.o isdigits/src/isdigits.cpp
    error: isdigits/src/isdigits.cpp:3:10: fatal error: string: No such file or directory
        3 | #include <string>
          |          ^~~~~~~~
    compilation terminated.

**Comment by waruqi** [comment]
> /home/raphy/.xmake/packages/c/cosmocc/3.4.0/7e5513266cda47b5b64419a0b92403ee/bin/cosmocc -c -fvisibility=hidden -DNDEBUG -o isdigits/builddir/.objs/isdigits/linux/x86_64/release/isdigits/src/isdigits.cpp.o isdigits/src/isdigits.cpp
error: isdigits/src/isdigits.cpp:3:10: fatal error: string: No such file or directory

This is an issue with cosmocc, I'm not sure if it supports stl and can find it automatically. xmake doesn't need to do anything else.

**Reply by raphael10-collab** [reply]
I opened an issue in the cosmopolitan's github page : https://github.com/jart/cosmopolitan/issues/1236
